hook/hooktest/hooktest.go

```diff
@@ -53,6 +63,9 @@ type Runner struct {
 //
 // Any hook tools that have been run will be stored in r.Record.
 func (runner *Runner) RunHook(hookName string, relId hook.RelationId, relUnit hook.UnitId) error {
+	if runner.HookStateDir == "" {
+		panic("empty hook state dir")
+	}
```

Comment:
perhaps a check if runner.HookStateDir exists (os.Stat) and is writable (syscall.Access)?